proxy-recorder/process_gatling_recording.rb

```diff
@@ -221,55 +237,110 @@ def step5_update_extends(text)
   text.gsub!(/^class (.*) extends Simulation/, 'class \1 extends SimulationWithScenario')
 end
 
-# Step 6
-def step6_comment_out_http_protocol(text)
-  puts "STEP 6: Comment out `httpProtocol` variable"
-  begin_comment = false
-  end_comment = false
+def step6_extract_http_protocol_headers(text)
+  puts 'STEP 6: Extract common headers from httpProtocol'
+  lines = text.lines
+  found_header_data = {}
+
+  http_protocol_start = lines.find_index {|line| line.match(/^.*val httpProtocol/)}
+
+  if http_protocol_start.nil?
+    puts "Can't find headers after httpProtocol definition"
+    exit 1
+  end
+
+  # +2 to skip .baseURL call
+  index = http_protocol_start + 2
```

Comment:
Maybe it would be better to just skip the first line rather than the first two - e.g., in the event that the `.baseURL` call were to start appearing somewhere below the second line in the future, like:

```scala
 val httpProtocol = http
 	.acceptHeader("application/json, text/pson")
 	.baseURL("https://puppetserver-perf-sut57.delivery.puppetlabs.net:8140")
 	...
```

If we did that and changed the logic to only add an entry into the `found_header_data` hash if the `header_name` is actually found in the `HELPER_TO_HEADER` hash, then I think we'd get the same result?